src/lib/operators/table_scan.cpp

```
@@ -0,0 +1,241 @@
+#include "table_scan.hpp"
+
+#include <functional>
+#include <utility>
+#include <memory>
+#include <vector>
+
+#include "resolve_type.hpp"
+#include "storage/chunk.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/table.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+namespace {
+template <typename T>
+std::function<bool(const T&, const T&)> make_comparator(opossum::ScanType scan_type) {
+  using Type = opossum::ScanType;
+  switch (scan_type) {
+    case Type::OpEquals:
+      return std::equal_to<T>();
+    case Type::OpNotEquals:
+      return std::not_equal_to<T>();
+    case Type::OpLessThan:
+      return std::less<T>();
+    case Type::OpLessThanEquals:
+      return std::less_equal<T>();
+    case Type::OpGreaterThan:
+      return std::greater<T>();
+    case Type::OpGreaterThanEquals:
+      return std::greater_equal<T>();
+  }
+  DebugAssert(false, "Operator not Implemented");
+  return [](T t1, T t2) { return false; };
```

you might want to throw an exception here instead, since this should never happen